lib/origen_arm_debug/mem_ap.rb

```diff
@@ -100,6 +100,12 @@ def write_register(reg_or_val, options = {});
         addr = reg_or_val                 # if not a register, use the 'val' as target addr
         wdata = options[:wdata]
       end
+
+      if reg_or_val.is_a? Numeric  # if not a register or value, the calling application generates the addr and register data to use and passes it as options to the method.
+        addr = options[:reg_addr]
+        wdata = options[:reg_data]
```

Comment:
I'm going to go ahead and merge this, but I'm going to convert everything (in the next few days) to use the standard (required arg=data, :address option in case of value).